components/frame-jsonld.js

```diff
@@ -0,0 +1,35 @@
+// frame-jsonld.js
+
+var _ = require('underscore');
+var jsonld = require('jsonld').promises;
+
+var wrapper = require('../src/javascript-wrapper.js');
+
+/**
+ * Converts the given JSON object into a JSON LD Graph object using an optional
+ * frame and context. If a frame is provided it will be used to frame the json
+ * object, parsed using the optional context. If a frame is not provided, but a
+ * context object is provided, it will be used to compact  the json
+ * object, parsed using the context if the json object doesn't have one.
+ * Otherwise, the json object is returned.
+ * @param input RDF JS Interface Graph object
+ * @param frame JSON-LD Frame object
+ * @param context JSON-LD Context object
+ */
+module.exports = wrapper(function execute(json, frame, context) {
```

Comment:
For stylistic reasons, I would rather that the updater ('execute' in this case) be declared outside of the wrapper call.